tests/core/component.test.js

```diff
@@ -225,4 +225,30 @@ suite('Component', function () {
       assert.ok(component.schema.color);
     });
   });
+
+  suite('updateProperties', function () {
+    setup(function () {
+      components.dummy = undefined;
+    });
+
+    test('updates the schema with the new value', function () {
+      var TestComponent = registerComponent('dummy', {
+        schema: { color: { default: 'red' } }
+      });
+      var el = document.createElement('a-entity');
+      var component = new TestComponent(el);
+      component.updateProperties(null);
+      assert.equal(component.data.color, 'red');
+    });
+
+    test('updates the schema with a null value', function () {
+      var TestComponent = registerComponent('dummy', {
+        schema: { color: { default: 'red' } }
```

Comment:
going forward, should we stick with spaces-in-objects style? can visually confuse it with block braces, not consistent with arrays, less compact